components/base/agilex/limo_base.go

```diff
@@ -330,12 +330,19 @@ func (lb *limoBase) setMotionCommand(linearVel float64,
 	return nil
 }
 
+// positive angleDeg spins base left. degsPerSec is a positive angular velocity.
 func (lb *limoBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
 	lb.logger.Debugf("Spin(%f, %f)", angleDeg, degsPerSec)
 	secsToRun := math.Abs(angleDeg / degsPerSec)
 	var err error
 	if lb.driveMode == DIFFERENTIAL.String() || lb.driveMode == OMNI.String() {
-		err = lb.SetVelocity(ctx, r3.Vector{}, r3.Vector{Z: degsPerSec}, extra)
+		sign := 0.0
+		if angleDeg < 0 {
+			sign = -1.0
+		} else {
+			sign = 1.0
+		}
+		err = lb.SetVelocity(ctx, r3.Vector{}, r3.Vector{Z: sign * degsPerSec}, extra)
```

Comment:
degsPerSec can be negative. If it is important to match the comment you have up there, take the absolute value before you pass it in here.

Reply:
We don't want to take the absolute value here since we want the argument to set velocity to be negative.  The comment you reference is referring to the argument as it is passed in rather than the local variable

Reply:
I think we are confused about how Spin and SetVelocity should be called.  I was under the assumption Spin accepted a signed degrees and an unsigned velocity.  And that SetVelocity accepted a signed velocity.  Is this correct?

Reply:
I chose to make it that angleDeg changes the direction of motion and degsPerSec is always positive because the wheeled_base driver operates in that way.